raft-tech/TANF-app
# Feat/1439 separate rds databases

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1439 

## How to Test
_List the steps to test the PR_

You can verify that these databases exist by installing and using this plugin:
```
$ cf login -a api.fr.cloud.gov --sso
$ cf install-plugin https://github.com/cloud-gov/cf-service-connect/releases/download/1.1.0/cf-service-connect-linux-386
$ cf connect-to-service tdp-backend-sandbox tdp-db-dev
Finding the service instance details...
Setting up SSH tunnel...
SSH tunnel created.
Connecting client...
psql (12.9 (Ubuntu 12.9-0ubuntu0.20.04.1), server 12.8)
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

cgawsbrokerprodx9hbe1uy1srystc=> \l
                                                        List of databases
              Name              |      Owner       | Encoding |   Collate   |    Ctype    |           Access privileges           
--------------------------------+------------------+----------+-------------+-------------+---------------------------------------
 cgawsbrokerprodx9hbe1uy1srystc | ubinm0tax9919ama | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 postgres                       | ubinm0tax9919ama | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 rdsadmin                       | rdsadmin         | UTF8     | en_US.UTF-8 | en_US.UTF-8 | rdsadmin=CTc/rdsadmin
 tdp_db_dev_sandbox             | ubinm0tax9919ama | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 template0                      | rdsadmin         | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/rdsadmin                          +
                                |                  |          |             |             | rdsadmin=CTc/rdsadmin
 template1                      | ubinm0tax9919ama | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/ubinm0tax9919ama                  +
                                |                  |          |             |             | ubinm0tax9919ama=CTc/ubinm0tax9919ama
(6 rows)

cgawsbrokerprodx9hbe1uy1srystc=> 
```

1. Add any of the Deploy github labels and verify this branch deploys and runs.
2. Log into the newly deployed environment and verify the database contents do not contain the same users/settings as other dev environments.
3. Check cloud.gov logs to ensure the initialization went without error.
> *Demo GIF(s) and screenshots for testing procedure*

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Each dev-space environment should have a separate database name within the shared tdp-db-dev RDS service.
- [x] Migrations/deployments in one dev space should not impact another's database.
- [x] The current users/roles data we have should remain a default through a Django migration.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Backend coverage: [_96.05%_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)


_n/a_ 

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):

_n/a_
